### PR TITLE
libct/cg: refactor/improve/rename GetHugePageSize -> HugePageSizes

### PR DESCRIPTION
**Pull request by kolyshkin**
~~_(this is based on and currently includes PR #3233)_~~

1. Since GetHugePageSize do not have any external users (checked by
       sourcegraph), and no internal user ever uses its second return value
       (the error), let's drop it.
    
2. Rename GetHugePageSize -> HugePageSizes (drop the Get prefix as per
       Go guidelines, add suffix since we return many sizes).

3.  Implement lazy init for `HugePageSizes`
    
    I have noticed that libct/cg/fs allocates 8K during init on every runc
    execution:
    
    > init github.com/opencontainers/runc/libcontainer/cgroups/fs @1.5 ms, 0.028 ms clock, **8512 bytes, 13 allocs**
    
    Apparently this is caused by the global `HugePageSizes` variable initialization,
    which is only used from `GetStat`s (i.e. it is never used by the runc binary itself).
    
    Remove it, and use `HugePageSizes()` directly instead. Make it init-once,
    so that `GetStats` (which, I guess, is periodically called by kubernetes)
    does not re-read huge page sizes over and over.
    
    This also removes 12 allocs and 8K from libct/cg/fs init section:
    
    > $ time GODEBUG=inittrace=1 ./runc --help 2>&1 | grep cgroups/fs
    > init github.com/opencontainers/runc/libcontainer/cgroups/fs @1.5 ms, 0.003 ms clock, **16 bytes, 1 allocs**

4. libct/cg: TestGetHugePageSizeImpl: use t.Run
    
    Move test case comments to doc strings, and use t.Run.

### CODE REVIEW

**Comment by kolyshkin** [comment]
This is definitely post-1.1 material

**Comment by kolyshkin** [comment]
Rebased on top of current git tip to repro https://github.com/opencontainers/runc/issues/3266

File: libcontainer/cgroups/utils_test.go [hunk 1]
```diff
@@ -487,25 +485,29 @@ func TestGetHugePageSizeImpl(t *testing.T) {
...
 		},
-		{ // no suffix (skipped with a warning)
-			input:  []string{"hugepages-1024"},
-			isWarn: true,
+		{ // mixed valid and invalid entries
+			input:  []string{"hugepages-4194304kB", "hugepages-2048kB", "hugepages-akB", "hugepages-64kB"},
+			output: []string{"4GB", "2MB", "64KB"},
+			isErr:  true,
+		},
+		{ // mixed valid and invalid entries
```
Comment (thaJeztah) [inline libcontainer/cgroups/utils_test.go:501]: No need to change, but could include the comment as a `doc: "mixed valid and invalid entries",`, and then use subtests;

```
for _, c := range testCases {
    c :=c
    t.Run(c.doc, func(t *testing.T) {
...
```

That way the test output is documenting what test fails (easier to find back which test is failing), (obvious disclaimer that I like subtests for that reason 😅)

**Review by thaJeztah — APPROVED** [review]
code LGTM

is the failure still the thing we're debugging?

File: libcontainer/cgroups/utils_test.go (hunk 1 above)
Comment (kolyshkin) [reply libcontainer/cgroups/utils_test.go]: Makes sense, added (as a separate commit for clarity and so you don't have to re-review the whole set).

**Comment by kolyshkin** [comment]
> is the failure still the thing we're debugging?

No, it was a one time thing, fixed by https://github.com/opencontainers/runc/pull/3273

**Comment by kolyshkin** [comment]
Since 1.1 is delayed I think this can be included.

**Review by thaJeztah — APPROVED** [review]
LGTM!

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**